qunicorn_core/core/pilotmanager/ibm_pilot.py

```diff
@@ -268,20 +273,32 @@ def __get_runtime_service(job: JobDataclass, token: Optional[str]) -> QiskitRunt
 
     @staticmethod
     def __map_runner_results_to_dataclass(
-        ibm_result: Result, job: JobDataclass, programs: Optional[Sequence[QuantumProgramDataclass]] = None
+        ibm_result: Result,
+        job: JobDataclass,
+        programs: Optional[Sequence[QuantumProgramDataclass]] = None,
+        circuits: List[QuantumCircuit] = None,
     ) -> list[ResultDataclass]:
         result_dtos: list[ResultDataclass] = []
+        classical_registers_metadata = []
+
+        for circuit in circuits:
+            for reg in circuit.cregs:
+                # FIXME: don't append registers that are not measured
```

Comment:
Ideally we implement the openqasm3 spec for output as mutch as possible (<https://openqasm.com/language/directives.html#input-output>). This would not depend on whether a rgeister was measured to...
To check all measurements use `circ.get_instructions("measure")` to get all measure instructions.
But this can be left for later (if so, we should create a tracking issue for this).